### DISCUSSION TITLE
F# example (Wolfenstein 3D)

### DISCUSSION THREAD

**Discussion by JamesRandall** [dotnet/Silk.NET#957] [Show and tell]
Don't know if its of interest to anyone but I just published some source code for a F# port of Wolfenstein 3D. Its a "traditional" raycaster and renders to the browser using Fable and canvas but their is also a native version that uses Silk .NET to raycast to a texture and then render that using OpenGL via Silk and plays WAV files. So quite simple from a Silk perspective but it does show the basics you need to do more (the actual code I used I lifted from a virtual console I work on from time to time).

Silk specific code can be found in the _desktop_ folder.

https://github.com/JamesRandall/fsharp-wolfenstein

**Comment by Perksey** [comment]
This is awesome!
